tests/migrations/test_operations.py

```diff
@@ -1632,6 +1632,45 @@ def test_rename_field(self):
         self.assertEqual(definition[1], [])
         self.assertEqual(definition[2], {'model_name': "Pony", 'old_name': "pink", 'new_name': "blue"})
 
+    def test_rename_field_with_db_column_defined(self):
+        """
+        RenameField with a db_column defined is a noop migration.
+        """
+        project_state = self.set_up_test_model("test_rnfl", unique_together=True, index_together=True)
+
+        # Add db_column to field, pink column is always pink column.
+        operations = [migrations.AlterField(
+            "Pony", "pink", field=models.IntegerField(default=3, db_column="pink"))]
+        self.apply_operations("test_rnfl", project_state, operations=operations)
+
+        # Test the state alteration
+        operation = migrations.RenameField("Pony", "pink", "blue")
+        new_state = project_state.clone()
+        operation.state_forwards("test_rnfl", new_state)
+
+        # Model changed
+        self.assertIn("blue", new_state.models["test_rnfl", "pony"].fields)
+        self.assertNotIn("pink", new_state.models["test_rnfl", "pony"].fields)
+
+        # But database are still the same
+        self.assertColumnExists("test_rnfl_pony", "pink")
+        self.assertColumnNotExists("test_rnfl_pony", "blue")
+        with connection.schema_editor() as editor:
+            # Must be a noop.
+            with self.assertNumQueries(0):
+                operation.database_forwards("test_rnfl", editor, project_state, new_state)
+
+        self.assertColumnExists("test_rnfl_pony", "pink")
+        self.assertColumnNotExists("test_rnfl_pony", "blue")
+
+        with connection.schema_editor() as editor:
+            # Must be a noop.
+            with self.assertNumQueries(0):
+                operation.database_backwards("test_rnfl", editor, new_state, project_state)
+
+        self.assertColumnExists("test_rnfl_pony", "pink")
+        self.assertColumnNotExists("test_rnfl_pony", "blue")
```

Comment:
I think we can simplify this test:
```suggestion
        self.assertIn('renamed_field', new_state.models['test_rfwdbc', 'pony'].fields)
        self.assertNotIn('field', new_state.models['test_rfwdbc', 'pony'].fields)
        self.assertColumnExists('test_rfwdbc_pony', 'db_field')
        with connection.schema_editor() as editor:
            with self.assertNumQueries(0):
                operation.database_forwards('test_rfwdbc', editor, project_state, new_state)
        self.assertColumnExists('test_rfwdbc_pony', 'db_field')
        with connection.schema_editor() as editor:
            with self.assertNumQueries(0):
                operation.database_backwards('test_rfwdbc', editor, new_state, project_state)
        self.assertColumnExists('test_rfwdbc_pony', 'db_field')
```